proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+    create substore entry with
+        substorePrefix = SUBSTORE_PREFIX_NFT
+        key = nftID
+        value = encode(
+            schema = NFTStoreSchema,
+            object = {
+               "owner": address,
+               "attributesArray": []
+            }
+        )
 ```
 
-#### getNFTowner
+#### createUserEntry
 
-This function returns the owner of an NFT.
+```python
+def createUserEntry(
+    address: Address,
+    nftID: NFTID
+) -> None:
+
+    # This may fail if there is not enough fee left.
+    Fee.payFee(USER_ENTRY_INITIALIZATION_FEE)
+
+    create substore entry with
+        substorePrefix = SUBSTORE_PREFIX_USER
+        key = address + nftID
+        value = encode(
+            schema = userStoreSchema,
+            object = {
+               "lockingModule": NFT_NOT_LOCKED
+            }
+        )
+```
+
+#### createEscrowEntry
+
+```python
+def createEscrowEntry(
+    receivingChainID: ChainID,
+    nftID: NFTID
+) -> None:
+
+    # This may fail if there is not enough fee left.
+    Fee.payFee(ESCROW_ENTRY_INITIALIZATION_FEE)
+
+    create substore entry with
+        substorePrefix = SUBSTORE_PREFIX_ESCROW
+        key = receivingChainID + nftID
+        value = encode(
+            schema = escrowStoreSchema,
+            object = {
+                "escrow": True
+            }
+        )
+```
+
+#### transferInternal
+
+```python
+def transferInternal(
+    senderAddress: Address,
+    recipientAddress: Address,
+    nftID: NFTID
+) -> None:
+
+    delete entry userStore(senderAddress, nftID) from the user substore
+    createUserEntry(recipientAddress, nftID)
+    NFTStore(nftID).owner = recipientAddress
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_TRANSFER,
+        data = {
+            "senderAddress": senderAddress,
+            "recipientAddress": recipientAddress,
+            "nftID": nftID,
+            "result": RESULT_SUCCESSFUL
+        },
+        topics = [senderAddress, recipientAddress]
+    )
+```
+
+#### transferCrossChainInternal
+
+```python
+def transferCrossChainInternal(
+    senderAddress: Address,
+    recipientAddress: Address,
+    nftID: NFTID,
+    receivingChainID: ChainID,
+    messageFee: uint64,
+    data: str
+) -> None:
+
+    delete entry userStore(senderAddress, nftID) from the user substore
+
+    if getChainID(nftID) == OWN_CHAIN_ID:
+        NFTStore(nftID).owner = receivingChainID
+
+    if getChainID(nftID) == receivingChainID:
+        delete entry NFTStore(nftID) from the NFT substore
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_TRANSFER_CROSS_CHAIN,
+        data = {
+            "senderAddress": senderAddress,
+            "recipientAddress": recipientAddress,
+            "nftID": nftID,
+            "receivingChainID": receivingChainID,
+            "result": RESULT_SUCCESSFUL
+        },
+        topics = [senderAddress, recipientAddress, receivingChainID]
+    )
+
+    Interoperability.send(
```

Comment:
We modified interop.send and now the params input must be of type bytes (you just need to encode it before sending it)